nixos/modules/services/web-apps/matomo.nix

```diff
@@ -169,8 +187,37 @@ in {
       '';
     };
 
+    # If this is run regularly via the timer,
+    # 'Browser trigger archiving' can be disabled in Matomo UI > Settings > General Settings.
+    systemd.services."matomo-archive-processing" = {
+      description = "Archive Matomo reports";
+      # the archiving can only work if the database is already up and running
+      requires = [ databaseService ];
+      after = [ databaseService ];
+
+      serviceConfig = {
+        Type = "oneshot";
+        User = user;
+        UMask = "0007";
+        # TODO: might get renamed to MATOMO_USER_PATH in future versions
+        Environment = "PIWIK_USER_PATH=${dataDir}";
+        ExecStart = "${cfg.package}/bin/matomo-console core:archive --url=https://${user}.${fqdn}";
+      };
+    };
+
+    systemd.timers."matomo-archive-processing" = mkIf cfg.periodicArchiveProcessing {
+      description = "Automatically archive Matomo reports every hour";
+
+      wantedBy = [ "timers.target" ];
```

Comment:
I always thought a `partOf = [ "matomo-archive-processing.service" ];` was needed too, but I guess not.